src/classes/BDI_DataImportService.cls

```diff
@@ -721,9 +721,39 @@ global with sharing class BDI_DataImportService {
                 }
             }
         }
+        if (isCurrencyMapped(strBaseObj)) {
+            addCurrencyIsoCode(mapDIFieldToObjField);
+        }
+
         return mapDIFieldToObjField;
     }
 
+    /*******************************************************************************************************
+    * @description Checks if this is one of the objects where we want to have a currency mapped.
+    * As of now, those objects are Opportunity and Payments. Payment currency iso code is inherited from the Opportunity.
+    * @param strBaseObj the underlying object the bdi object is (ie, Contact, Account, etc)
+    * @return Boolean if the strBaseObj is either a Payment or an Opportunity.
+    */
+    private static Boolean isCurrencyMapped(String objectName) {
+        Set<String> objectsThatMapCurrencyCode = new Set<String>{
```

Comment:
At this point we just simplify this to `return objectName == 'Opportunity'`. I still like leaving it as a separate method in case we want to expand in the future.